src/components/AuthContextProvider.js

```diff
@@ -1,34 +1,58 @@
 import { createContext, useContext, useState } from 'react';
-
 export const AuthContext = createContext();
 export const useAuthContext = () => useContext(AuthContext);
 import { login } from '../services/auth';
 import { userRegister } from '../services/register';
 import { registerCompany } from '../services/company';
 import { uploadPhoto } from '../services/upload';
-import { createProfile } from '../services/profile';
+import { createProfile, getProfile } from '../services/profile';
+import { getUserInfo } from '../services/user';
+import { jwtInterceptor } from '../services/http';
 
 const AuthContextProvider = ({ children }) => {
   const [errorVisible, setErrorVisible] = useState(false);
-  const [jwt, setJwt] = useState();
+  const [jwt, setJwt] = useState(window.localStorage.getItem('jwt'));
+  const [userName, setUserName] = useState();
+  const [profilePhoto, setProfilePhoto] = useState();
   const [userData, setUserData] = useState();
-  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(jwt ? true : false);
   const [activeOption, setActiveOption] = useState('login');
 
   const handleLogout = () => {
+    window.localStorage.removeItem('jwt');
+    window.localStorage.removeItem('userName');
+    window.localStorage.removeItem('profilePhoto');
+    setUserName('');
+    setProfilePhoto('');
     setIsLoggedIn(false);
     setActiveOption('login');
+    location.reload();
   };
 
   const handleLogin = async (event, email, password) => {
     event.preventDefault();
     try {
       const authenticatedUser = await login(email, password);
+      setUserData(authenticatedUser.data);
       setJwt(authenticatedUser.jwt);
-      setUserData(authenticatedUser.user);
-      setIsLoggedIn(true);
+      window.localStorage.setItem('jwt', authenticatedUser.jwt);
       setActiveOption(null);
       setErrorVisible(false);
+
+      jwtInterceptor(authenticatedUser.jwt);
```

Comment:
1. We should not call this here for a few reasons - if we call this function multiple times, we'll add multiple interceptors with the same function, and we want to call it when the app first initialized just once, so it adds the interceptor in the case we're already logged in
2. Don't pass in the JWT value to the function, but have the interceptor function read the token from localstorage for every request - that way we'll catch the token changes (e.g. you login, logout, login again, and this way you'll always send the most recent token value)
3. So a better solution would be to execute `jwtInterceptor` in `AuthContext` (once when the app first starts)